test/integration/gnosisSafeNestedSafes.spec.ts

```diff
@@ -0,0 +1,123 @@
+
+        const ownersSignSafe1 =  buildSignatureBytes([sig1, sig2])
+        const ownersSignSafe2 =  buildSignatureBytes([sig3, sig4])
+        
+        // Check if signature for each safe is correct
+        expect(await handlerSafe1.callStatic['isValidSignature(bytes,bytes)'](messageData, ownersSignSafe1)).to.be.eq("0x20c13b0b")
+        expect(await handlerSafe2.callStatic['isValidSignature(bytes,bytes)'](messageData, ownersSignSafe2)).to.be.eq("0x20c13b0b")
+
+        const signerSafe1 = buildContractSignature(safe1.address, ownersSignSafe1)
+        const signerSafe2 = buildContractSignature(safe2.address, ownersSignSafe2)
+        const signature = buildSignatureBytes([signerSafe1, signerSafe2])
+        // Should execute transaction withdraw 1 ether
+        expect(await parentSafe.execTransaction(to, value, data, operation, 0, 0, 0, AddressZero, AddressZero, signature)).to.be.ok
+        
+        // Should be 0 
+        await expect(await hre.ethers.provider.getBalance(parentSafe.address)).to.be.deep.eq(parseEther("0"))
+    })
+
+    it('should revert with hash not approved ', async () => {
+        const { safe1, safe2, parentSafe, handlerSafe1, signLib} = await setupTests()
+        // Deposit some spare money for execution to parent safe
+        await expect(await hre.ethers.provider.getBalance(parentSafe.address)).to.be.equal(0)
+        await user1.sendTransaction({to: parentSafe.address, value: parseEther("1")})
+        await expect(await hre.ethers.provider.getBalance(parentSafe.address)).to.be.equal(parseEther("1"))
+
+        // Withdraw 1 ETH
+        const to = user5.address
+        const value = parseEther("1")
+        const data = "0x"
+        const operation = 0
+        const nonce = await parentSafe.nonce()
+        const messageData = await parentSafe.encodeTransactionData(to, value, data, operation, 0, 0, 0, AddressZero, AddressZero, nonce)
+        
+        // Get hash transaction for each safe
+        const messageHashSafe1 = await handlerSafe1.getMessageHashForSafe(safe1.address, messageData)
+
+        // Get all signs for each owner Safe1 (user1, user2) Safe2 (user3, user4)
+        const sig1 = await signHash(user1, messageHashSafe1)
+        const sig2 = await signHash(user2, messageHashSafe1)
+        const ownersSignSafe1 =  buildSignatureBytes([sig1, sig2])
+         
+        // Check if signature for each safe is correct
+        expect(await handlerSafe1.callStatic['isValidSignature(bytes,bytes)'](messageData, ownersSignSafe1)).to.be.eq("0x20c13b0b")
+
+        const signerSafe1 = buildContractSignature(safe1.address, ownersSignSafe1)
+        //Create an empty signature to with dynamic true to include this address in the staticpart
+        const emptySignSafe2: SafeSignature = {
+            signer: safe2.address,
+            data: "",
+            dynamic: true,
+        };
+        const signature = buildSignatureBytes([signerSafe1,emptySignSafe2])
```

Comment:
@mikhailxyz, `emptySignSafe2` was created to include the address of safe2 in the `staticpart`.